storagenode/piecestore/endpoint.go

```diff
@@ -352,6 +360,11 @@ func (endpoint *Endpoint) doUpload(stream uploadStream) (err error) {
 				return ErrInternal.Wrap(err)
 			}
 
+			// Save the order before completing the call. Set orderSaved so
+			// that the defer above does not also save.
+			orderSaved = true
+			endpoint.saveOrder(ctx, limit, &largestOrder)
```

Comment:
It does not return an error, because was done in a defer before so the error is simply logged. The `orderSaved = true` simply prevents two _attempts_ to save from occurring, but neither guarantee the save actually succeeds. I agree this is a problem that should be fixed, but I'd prefer to do that in a separate ticket.

Reply:
👍